src/resolvers/profileResolver.ts

```diff
@@ -0,0 +1,54 @@
+import { Profile } from '../models/user'
+import cloudinary from 'cloudinary';
+import path from 'path';
+
+cloudinary.config({
+  cloud_name: process.env.CLOUD_NAME,
+  api_key: process.env.API_KEY,
+  api_secret: process.env.API_SECRET
+});
+
+const profileResolvers = {
+    Query: {
+        getProfile: (parent:any , args: any) => {
+            return Profile.findById(args.id)
+        }
+    },
+    Mutation: {
+        updateProfile: async (parent:any, args:any, context: any) => {
+            try {
+              const { id, firstName, lastName, address, city, country, phoneNumber, biography, fileName } = args
+
+              const { userId } = context;
+
+                if (!userId) {
+                    throw new Error("You need to login first")
+                }
+
+                if (fileName)  {
+                  const mainDir = path.dirname(require.main?.filename)
+                  fileName = `${mainDir}/uploads/${fileName}`                
+                  const photo = await cloudinary.v2.uploader.upload(fileName)
+                }
+                
+                
+              const updatedProfile = await Profile.findByIdAndUpdate(id, { 
+                firstName: firstName,
+                lastName: lastName,
+                address: address,
+                city: city,
+                country: country,
+                phoneNumber: phoneNumber,
+                biography: biography,
+                avatar: photo?.secure_url
```

Comment:
you will not have access to the `photo` variable since it was defined in the `if` block which tie the `photo` to that block only.

Please adjust